### DISCUSSION TITLE
Resources names without magic strings?

### DISCUSSION THREAD

**Discussion by TonyK-FmGroup** [unoplatform/uno#8546] [Q&A]
Using the resw files, the docs show examples of getting the content using "string_literals" but I'd much prefer something a bit more compile safe.
Is it possible to get the names from the resources as something more concrete such as properties? Something that will give an error if it doesn't exist at compile time when I use them in code?

For example:

```
var text = GetString(Resources.FilterName);
```

**Comment by pkar70** [comment]
It can also be "/file/name", see https://github.com/unoplatform/uno/pull/2050 (merged).

**Reply by TonyK-FmGroup** [reply]
Thanks, but this looks like it also still needs to be a string literal which wouldn't be caught at compile time. Can we do it in a way that will catch missing items at compile time?

**Reply by jeromelaban** [reply]
Resw files don't have support for compile-time members. You can use resx files instead, but those cannot be used from XAML.

**Comment by pkar70** [comment]
It is hard to detect (detect if item is missing). Should compiler checks all items in all cultures (all resw files)? Or only in fallback culture? What about e.g. `GetString("q" + itemNumber.ToString())`?

**Comment by DierkDroth** [comment]
FWIW

I ended up
- only doing RESX files and leaving RESW files in libraries and the sahred project
- writing a tool (not using the standard generator tool) which builds C# wrapper files for the RESX files (which basically are just XML files)

[Details here](https://github.com/unoplatform/uno/discussions/8407)

**Reply by TonyK-FmGroup** [reply]
Thank you, I'll take a look.  But, does this approach still allow for the x:Uid approach to using the localized names in XAML?

**Reply by DierkDroth** [reply]
Sorry, I would not know since I'm not doing any XAML but doing all my UI in C# with an approach similar to [here](https://github.com/VincentH-Net/CSharpForMarkup)
